utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -324,8 +337,8 @@ def get_linux_sysroot(self, platform, arch):
         return os.path.join(os.sep, 'usr', sysroot_dirname)
 
     def get_linux_target(self, platform, arch):
-        sysroot_arch, abi = self.get_linux_abi(arch)
-        return '{}-unknown-linux-{}'.format(sysroot_arch, abi)
+        sysroot_arch, distro, abi = self.get_linux_abi(arch)
+        return '{}-{}-linux-{}'.format(sysroot_arch, distro, abi)
```

Comment:
The field being substituted is the "vendor".  I agree with @etcwilde that we should stick to standard terminology here.

SuSe also depends on the vendor being proper IIRC.  Keep in mind that this will make cross-compilation more complicated as the target triple is embedded into the serialised swiftmodule.  Do we want to consider filtering this in the driver?  That would allow us to use the same swiftmodule on different distributions but get the correct triple for the link phase.

It may be interesting to consider actually adding a new field into the swiftmodule to actually embed a different clang triple to the swift triple.

Reply:
> The field being substituted is the "vendor". I agree with @etcwilde that we should stick to standard terminology here.

Thanks, that's updated now.

> SuSe also depends on the vendor being proper IIRC. Keep in mind that this will make cross-compilation more complicated as the target triple is embedded into the serialised swiftmodule. Do we want to consider filtering this in the driver? That would allow us to use the same swiftmodule on different distributions but get the correct triple for the link phase.
> 
> It may be interesting to consider actually adding a new field into the swiftmodule to actually embed a different clang triple to the swift triple.

I don't have an informed opinion on this. Should I tag someone else to review this aspect of the change?

Reply:
I could have sworn that the name of the swift module helped in this regard. e.g. `Blarpy.swiftmodule/x86_64-apple-macos.swiftmodule` vs `Blarpy.swiftmodule/arm64-apple-macos.swiftmodule` are applied to their corresponding architectures on macOS. I haven't looked closely so take it with a grain of salt, but having the swiftmodule key'd on its full triple should work on Linux too. The build system probably needs to learn to do this on things that aren't just Apple platforms though.

@xymus, can you confirm that? Thanks.

Reply:
Having a different target triple should help to avoid loading incompatible swiftmodules in the case of different distributions. I wouldn't trust loading a swiftmodule built on a different distro as they could refer to things from the C world that are organized differently. That being said, I don't work much in the Linux side of things so I'd be interested to know if it's an accepted practice there.

Reply:
The default Linux toolchain distribution does not use the directory layout like Windows and macOS, so you lose that ability to differentiate.  How much C code actually gets serialised into the object layout?  The only case that I would actually be concerned about is if we are serialising **opaque** C structures (e.g. `pthread_t`) somehow.  The remainder of the differences I believe should be handled implicitly as the macros are contained within the C module, which is re-constructed through the modulemap.  The current loading behavior I feel may be a bit confusing for users.  We should definitely include a test case for that along with a reasonable error message if we intend to ship this as is.